pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1326,7 +1327,9 @@ protected void handleProducer(final CommandProducer cmdProducer) {
 
                 backlogQuotaCheckFuture.thenRun(() -> {
                     // Check whether the producer will publish encrypted messages or not
-                    if ((topic.isEncryptionRequired() || encryptionRequireOnProducer) && !isEncrypted) {
+                    if ((topic.isEncryptionRequired() || encryptionRequireOnProducer)
+                            && !isEncrypted
+                            && !SystemTopicNames.isSystemTopic(topicName)) {
```

Comment:
If `topic.isEncryptionRequired()` is always `false` for system topics, why do we need the check on the `topicName` on line 1332?

Reply:
the previous condition was: topic requires encryption or producer requested encryption 
if I leave it as is, the `encryptionRequireOnProducer` will be valid also for system topics and then the `opic.isEncryptionRequired()` would be useless

Reply:
Oh, of course. Thanks for explaining.